perfkitbenchmarker/providers/azure/azure_network.py

```diff
@@ -21,264 +21,284 @@
-                  'create',
-                  '--affinity-group=%s' % self.name,
-                  self.name]
-    vm_util.IssueCommand(create_cmd)
+    vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'network', 'vnet', 'create',
+         '--location', self.location,
+         self.name] + self.resource_group.args)
 
   def _Delete(self):
     """Deletes the virtual network."""
-    delete_cmd = [AZURE_PATH,
-                  'network',
-                  'vnet',
-                  'delete',
-                  '--quiet',
-                  self.name]
-    vm_util.IssueCommand(delete_cmd)
+    vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'network', 'vnet', 'delete',
+         '--quiet',
+         self.name] + self.resource_group.args)
 
   def _Exists(self):
     """Returns true if the virtual network exists."""
-    show_cmd = [AZURE_PATH,
-                'network',
-                'vnet',
-                'show',
-                '--json',
-                self.name]
-    stdout, _, _ = vm_util.IssueCommand(show_cmd, suppress_warning=True)
-    vnet = json.loads(stdout)
-    if vnet:
-      return True
-    return False
+    stdout, _, retcode = vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'network', 'vnet', 'show',
+         '--json',
+         self.name] + self.resource_group.args,
+        suppress_warning=True)
+
+    return retcode == 0 and stdout != '{}\n'
+
+
+class AzureSubnet(resource.BaseResource):
+  def __init__(self, vnet, name):
+    super(AzureSubnet, self).__init__()
+    self.resource_group = GetResourceGroup()
+    self.vnet = vnet
+    self.name = name
+    self.args = ['--vnet-subnet-name', self.name]
+
+  def _Create(self):
+    logging.info('subnet._Create')
+    vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'network', 'vnet', 'subnet', 'create',
+         '--vnet-name', self.vnet.name,
+         '--address-prefix', self.vnet.address_space,
+         self.name] + self.resource_group.args)
+
+  def _Exists(self):
+    stdout, _, retcode = vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'network', 'vnet', 'subnet', 'show',
+         '--vnet-name', self.vnet.name,
+         '--json',
+         self.name] + self.resource_group.args)
+
+    return retcode == 0 and stdout != '{}\n'
+
+  def _Delete(self):
+    vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'network', 'vnet', 'subnet', 'delete',
+         '--vnet-name', self.vnet.name,
+         '--quiet',
+         self.name] + self.resource_group.args)
+
+
+class AzureFirewall(network.BaseFirewall):
+  """An object representing the Azure Firewall equivalent.
+  """
+
+  CLOUD = providers.AZURE
+
+  def AllowPort(self, vm, port):
+    """Opens a port on the firewall.
+
+    Args:
+      vm: The BaseVirtualMachine object to open the port for.
+      port: The local port to open.
+    """
+    if vm.is_static or port == SSH_PORT:
+      return
+
+  def DisallowAllPorts(self):
+    """Closes all ports on the firewall."""
+    pass
```

Comment:
@voellm I'm planning to implement NSGs instead, so I won't have to ignore it. It looks to me from the docs that the typical architecture is an NSG per subnet, so that's my plan. Does that sound about right to you?